vignettes/cookbook_preparatory_steps.Rmd

```diff
@@ -14,6 +14,233 @@ knitr::opts_chunk$set(
+(2) `git` installed locally,
+(3) set up credentials so that RStudio can communicate with GitHub.
+
+You can find more information on how to do this using the following resources:
+
+* [Happy Git and GitHub for the useR](https://happygitwithr.com/) is a great and comprehensive resource that takes you through the process of setting up git and GitHub with RStudio, including registering a GitHub account, installing git, and connecting RStudio to GitHub.
+* Additional information on managing your GitHub connection from within RStudio can be found in the usethis package documentation, for example on [managing git credentials](https://usethis.r-lib.org/articles/git-credentials.html).
+
+If you only plan to use GitHub to install this package or other packages as shown above, you will not have to have a deep understanding of all the git commands, so there is no need to be overwhelmed by the complexity of git.
+
+
+### Required R packages
+
+The `r2dii.*` suite of R packages depends on a number of other R packages. These dependencies will be installed automatically when you install the `r2dii.*` suite of R packages. The required packages are:
+
+```{r imports, echo = FALSE, results = 'asis'}
+imports <- paste(
+  utils::packageDescription("r2dii.data")[["Imports"]],
+  utils::packageDescription("r2dii.match")[["Imports"]],
+  utils::packageDescription("r2dii.analysis")[["Imports"]],
+  utils::packageDescription("r2dii.plot")[["Imports"]],
+  sep = ", "
+)
+imports <- sort(unique(unlist(strsplit(x = imports, split = ",[[:space:]]"))))
+cat(paste0("`", paste0(imports, collapse = "`, `"), "`"))
+```
+
+### Suggested R packages
+
+The suggested packages are not required to run the analysis, but they are used in the examples and vignettes provided with the package:
+
+```{r suggests, echo = FALSE, results = 'asis'}
+imports <- paste(
+  utils::packageDescription("r2dii.data")[["Suggests"]],
+  utils::packageDescription("r2dii.match")[["Suggests"]],
+  utils::packageDescription("r2dii.analysis")[["Suggests"]],
+  utils::packageDescription("r2dii.plot")[["Suggests"]],
+  sep = ", "
+)
+imports <- sort(unique(unlist(strsplit(x = imports, split = ",[[:space:]]"))))
+cat(paste0("`", paste0(imports, collapse = "`, `"), "`"))
+```
+
+### FAQ
+
+#### How do I install the `r2dii.*` suite of R packages?
+
+The most common ways to install R packages are via CRAN or [GitHub](#connecting-to-github-from-rstudio). Public institutions often have restrictions on the installation of packages from GitHub, so you may need to install the package from CRAN. In some cases, your institution may mirror CRAN in their internal application registry, so you may need to install the package from there. Should you have any issues with the installation from the internal application registry, it is best to reach out to your IT department. If you cannot obtain the package in any of these ways, please reach out to the package maintainers directly for exploring other options.
```

Comment:
again depends on whether or not we feel like mentioning github is worth it at all here, since `r2dii.*` is on CRAN

Reply:
seems like we decided to keep both release and dev install instructions